extensions/amp-lightbox-viewer/0.1/test/test-amp-lightbox-viewer.js

```diff
@@ -63,33 +63,44 @@ describes.realWin('amp-lightbox-viewer', {
   });
 
   function runTests(autoLightbox) {
-    it('should build', () => {
+    it('should build on activation', () => {
       return getAmpLightboxViewer(autoLightbox).then(viewer => {
-        const container = viewer.querySelector('.i-amphtml-lbv');
-        expect(container).to.exist;
+        const impl = viewer.implementation_;
+
+        impl.vsync_.mutate = function(callback) {
+          callback();
+        };
+        impl.resources_.requireLayout = function() {
+          return Promise.resolve();
+        };
+        expect(viewer.style.display).to.equal('none');
+        return impl.activate({source: item1}).then(() => {
```

Comment:
I think the better Promise style is to return the promise, then have the `then` clause on the outer chain.

```js
return getAmpLightboxViewer(autoLightbox).then(viewer => {
  /*...*/
  return impl.activate({source: item1});
})
.then(() => {
  const container = viewer.querySelector('.i-amphtml-lbv');
  /*...*/
});
```

That can help avoid the pyramid shaped nesting getting deeper

> ["You can easily smell a promise pyramid whenever you see then() blocks going inwards in an existing then() block."](http://blog.ministryofprogramming.com/prevent-pyramids-with-flat-promise-chains/)

Reply:
Thanks a lot for this. TIL!